src/main/asciidoc/envers.adoc

```diff
@@ -0,0 +1,134 @@
+[[envers.what]]
+== What is Spring Data Envers
+
+Spring Data Envers differs from other Spring Data modules in that it is always used in combination with another Spring Data Module: Spring Data JPA.
+It makes typical Envers queries available in repositories for Spring Data JPA.
+
+== What is Envers?
+
+Envers is a Hibernate module which adds auditing capabilities to JPA entities.
+This documentation assumes you are familiar with Envers just as Spring Data Envers relies on Envers being properly configured.
+
+
+[[envers.getting-started]]
+== Getting Started
+
+As a starting point for using Spring Data Envers you need a project with Spring Data JPA.
+The easiest way to create that is using the Spring Initializr either on start.spring.io or via its integration in your favorite IDE.
+
+Now add spring-data-envers as a dependency
+[source,xml,subs="+attributes"]
+----
+<dependencies>
+
+  <!-- other dependency elements omitted -->
+
+  <dependency>
+    <groupId>org.springframework.data</groupId>
+    <artifactId>spring-data-envers</artifactId>
+    <version>{version}</version>
+  </dependency>
+
+</dependencies>
+----
+This will also bring hibernate-envers into the project as a transient dependency.
+
+Next is to make one or more repositories into {spring-data-commons-javadoc-base}/org/springframework/data/repository/history/RevisionRepository.html[`RevisionRepository`] by adding it as an extended interface.
+
+[source,java]
+----
+public interface PersonRepository
+    extends CrudRepository<Person, Long>,
+    RevisionRepository<Person, Long, Long> {}
+----
+
+In order to provide an implementations contained in that interface we have to use a different `repositoryFactoryBeanClass` for creating repositories.
+
+[source,java]
+----
+@SpringBootApplication
+@EnableJpaRepositories(repositoryFactoryBeanClass = EnversRevisionRepositoryFactoryBean.class)
+public class EnversDemoConfiguration {}
+----
+
+The entity for that repository should be an entity with Envers auditing enabled, i.e. it has an `@Audited` annotation.
+
+[source,java]
+----
+@Entity
+@Audited
+public class Person {
+
+	@Id @GeneratedValue
+	Long id;
+	String name;
+	@Version Long version;
+}
+----
+
+You may now use the methods from `RevisionRepository` to query the revisions of the entity as demonstrated in the following test case.
+
+
+
+[source,java]
+----
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
```

Comment:
Can we rewrite the test to not require Spring Boot?

Reply:
Wouldn't that mean we have to reference an application context configuration which then should also be included in the example? Or are you aiming for something else?